src/psyclone/psyir/backend/fortran.py

```diff
@@ -1424,14 +1424,23 @@ def loop_node(self, node):
         start = self._visit(node.start_expr)
         stop = self._visit(node.stop_expr)
         step = self._visit(node.step_expr)
-        variable_name = node.variable.name
 
         self._depth += 1
         body = ""
         for child in node.loop_body:
             body += self._visit(child)
         self._depth -= 1
 
+        # A generation error is raised if variable is not defined. This
+        # happens in LFRic kernel that iterate over a domain.
+        try:
+            variable_name = node.variable.name
+        except GenerationError:
```

Comment:
I tried, for the tests there was a `self.iteration_space` attribute. BUT, it turns out this is only there because `test_psy_gen_domain_kernel` in `lfric_domain_kernels_test` passes a non-lowered DynLoop (or lower_to_lang doesn't do it yet). And apparently nothing in the tests uses this.
Anyway, the tests failed when a clean `psyir.Loop` node is given  (. I can't see a better way of implementing this - I could test for `node.variable != None`, but `Loop.variable` checks its consistency and crashes if `_variable` is None. So I would have to access `node._variable` directly, which I like even more than the original code.
Let me know what you prefer to do.